### ISSUE TITLE
"Unauthorized" message with non-administrator user.

### ISSUE THREAD

**Issue by nnielzz** [resoai/TileBoard#369]
Everytime I load Tileboard with my "normal" user (doesn't have administrator privileges), I get the "Unauthorized" error.

Now, I can still normally use everything I have on my Tileboard without issues. Also, I get the "Failed login attempt"-notification inside Home Assistant from time to time (might be due to a token expiration?).

Judging by the console, it looks like an API-related error.
![Unauthorized](https://user-images.githubusercontent.com/24573360/81470340-9a943380-91ea-11ea-83c7-a1ab1ee13f99.png)

**Comment by rchl** [comment]
It looks like it's this API message that triggers it:
```
event_type: "tileboard"
id: 2
type: "subscribe_events"
```
So user doesn't have access to subscribe to "tileboard" events.
It might be solvable on the HA side or maybe they are still fleshing out permission stuff. Maybe worth asking in their tracker.

**Comment by nnielzz** [comment]
> It looks like it's this API message that triggers it:
> 
> ```
> event_type: "tileboard"
> id: 2
> type: "subscribe_events"
> ```
> 
> So user doesn't have access to subscribe to "tileboard" events.
> It might be solvable on the HA side or maybe they are still fleshing out permission stuff. Maybe worth asking in their tracker.

Yup, you seem to be on point. I have forwarded this issue to the Home Assistant Dev Team.

A temporary solution for now is to either login with your "admin"-account or to add a long lasting token from your admin account.

Thanks for the help.

**Comment by nnielzz** [comment]
Update:

I filed the issue on the Home Assistant GitHub, but they aren't helping me out (saying "TileBoard isn't  supported with Home Assistant").

Seems like I'm back the square one. Can this issue please be fixed?

**Comment by rchl** [comment]
I don't know internals of HA enough to see how this could be fixed.

It's not necessarily Tileboard specific issue though.
The issue is that normal users have no permission to subscribe to custom events using `subscribe_events` message.

**Comment by grahamPegNetwork** [comment]
Interesting, I've been wondering about this error.

If it is as @rchl describes, non-admin users can't subscribe to (custom?) events, then one of two things need two happen:

1. HA needs to add a permission or ability for normal users to use subscriptions, and it needs to be documented on TB
2. TB needs to work around HA's limitation by A. checking for permission or B. catching and handling the error.

**Comment by rchl** [comment]
@nnielzz Can you link to the issue you've created?

(I might have replied there but don't remember anymore).

**Comment by nnielzz** [comment]
> @nnielzz Can you link to the issue you've created?
> 
> (I might have replied there but don't remember anymore).

Yup, as @grahamPegNetwork mentions, they consider it a vulnerability.
Here's the link to the issue, though it's closed:
[https://github.com/home-assistant/core/issues/36122](url)

**Comment by rchl** [comment]
I've replied on that issue asking for clarification but then I agree with what was said there that this would be a security issue as custom event could carry some secret/personal information.

All TileBoard could do here is probably just not complaining about that and documenting that using `tileboard` events is not supported for non-admin users.

Ideally, there would be something exposed in the HA API that would allow that. The way TileBoard uses those events is perfectly safe as those are its own events. It's not trying to snoop on something else.
